common/scala/src/main/scala/whisk/common/TransactionId.scala

```diff
@@ -108,13 +141,27 @@ case class TransactionId private (meta: TransactionMetadata) extends AnyVal {
    */
   def failed(from: AnyRef, startMarker: StartMarker, message: String = "", logLevel: LogLevel = WarningLevel)(
     implicit logging: Logging) = {
+
     val endMarker =
       LogMarkerToken(startMarker.startMarker.component, startMarker.startMarker.action, LoggingMarkers.error)
-    logging.emit(
-      logLevel,
-      this,
-      from,
-      createMessageWithMarker(message, LogMarker(endMarker, deltaToStart, Some(deltaToMarker(startMarker)))))
+
+    if (TransactionId.metricsLog) {
+      logging.emit(
+        logLevel,
+        this,
+        from,
+        createMessageWithMarker(message, LogMarker(endMarker, deltaToStart, Some(deltaToMarker(startMarker)))))
+    } else {
+      if (message.nonEmpty) {
+        logging.emit(logLevel, this, from, message)
+      }
+    }
+
+    if (TransactionId.metricsKamon) {
+      val endTime = Instant.now(Clock.systemUTC)
+      val metricValue = deltaToMarker(startMarker, endTime)
```

Comment:
`deltaToMarker` has a default parameter, already getting the correct endTime for you. No need to explicitly get it again.